src/mlpack/methods/reinforcement_learning/environment/ftn.hpp

```diff
@@ -0,0 +1,300 @@
+/**
+ * @file methods/reinforcement_learning/environment/ftn.hpp
+ * @author Nanubala Gnana Sai
+ *
+ * This file is an implementation of Fruit Tree Navigation (FTN) Task:
+ * https://github.com/RunzheYang/MORL/blob/master/synthetic/envs/fruit_tree.py
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_RL_ENVIRONMENT_FTN_HPP
+#define MLPACK_METHODS_RL_ENVIRONMENT_FTN_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack{
+namespace rl{
+
+/**
+ * Implementation of Fruit Tree Navigation Task. 
+ */
+class FTN
```

Comment:
Just repeating my pedantic comment on changing the class name to be more verbose ie.
```suggestion
class FruitTreeNavigation
```

Also, it would be prudent to rename the header file to have a similar verbose name so that anyone using the library can easily understand what ftn stands for! :)